src/App.js

```diff
@@ -2,355 +2,76 @@ import React from "react";
-    }
-    return result
-  };
-
-  getLastChar = (from) => {
-    return from.slice(-1);
-  };
-
-
-  roundedResult = (expression) => {
-    const digitAfterComma = 13;
-    round(evaluate(expression), digitAfterComma);
-  };
-
-  addMultiplier = (expression) => {
-    //Find all pattern Digit + (
-    //expression ="1(2+6)+23(6+9)+(2+3)9"
-
-    const digitAndP = /([0123456789])(\()/g;
-    const pAndDigit = /(\))([0123456789])/g;
-    // )( -> )*(
-    const pAndp = /(\))(\()/g;
-
-    //return the value
-    return expression
-      .replace(digitAndP, "$1*$2")
-      .replace(pAndDigit, "$1*$2")
-      .replace(pAndp, "$1*$2");
-
-
-  };
-
-
-
-  calculate = () => {
-    const { result, operate, originalLastNum } = this.state;
-    let finalResult = 0;
-    var tempResult = result;
+      if (lastEquation !== "" && button.key !== "=") {
+        //start new equation
+        tempResult = "";
+      }
 
-    // Special case with expression ending with operator, after CE use
-    if ("-+*/".includes(this.getLastChar(result))) {
-      tempResult = result.slice(0, -1);
+      this.setState({
+        result: tempResult + currentBtn,
+        lastEquation: "",
+      });
     }
 
+    if (button.key === "=") {
+      var nextEquation = cleanupEquation(result, lastEquation);
 
-    tempResult = this.closeParens(tempResult) + "";
-    tempResult = this.addMultiplier(tempResult) + "";
-
-
-    // Handling double equal
-    // If I only have sign + digit(s) + dot + digit(s)
-    const regex = /-{0,1}[0123456789]*(\.[0123456789]*){0,1}/g;
-    const matches = tempResult.match(regex) || [];
-    if (matches.length > 1 && matches[0] === tempResult) {
-      //I need to handle the double equal
-      if ("+-/*".includes(operate) && !isNaN(originalLastNum)) {
-        // finalResult = evaluate(result + operate + originalLastNum) + "";
-        try {
-          finalResult = this.roundedResult(tempResult + operate + originalLastNum) + "";
-        } catch(error) {
-
-          //throw error
-        }
+      if (lastEquation.length > 0) {
+        nextEquation = repeatLastEquation(lastEquation, nextEquation);
       }
-    } else {
-      var tempResultString = tempResult;
-      //We need to detect the non nested parenthesis
-      //Open ( and no other ( before the next close
-      // '(' 0123456789-+*/.(not (, not ) ) ')'
-      // ( + any number of the 15 chars + ) : (7+2) (-3.25478/+6.587)
-      const reg = /\(([0123456789/*-+.]*)\)/g;
-      var parenthesisToCalculate = tempResultString.match(reg) || [];
-      while (parenthesisToCalculate.length > 0) {
-        for (var i = 0; i < parenthesisToCalculate.length; i++) {
-          //we extract the first expression ex:  (2+5)
-          let expression = parenthesisToCalculate[i];
-          //We calculate the value ex: 7
 
-          try {
-            let tempResult = round(evaluate(expression), 13);
+      let finalResult = 0;
+      [nextEquation, finalResult] = runEquation(nextEquation);
```

Comment:
@grimxreaper Can we extract everything in this equals pressed if block, excluding the setState, to a function? What would be a good name for it?